### PR TITLE
fix(form): use CSS vars for defining label cursor

### PR DESCRIPTION
**Pull request by dgdavid**
Allowing PatternFly users to customize the cursor shown when hovering a
form label.

Refs: https://github.com/patternfly/patternfly/issues/4777

### CODE REVIEW

File: src/patternfly/components/Form/form.scss [hunk 1]
```diff
@@ -67,6 +67,8 @@ $pf-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "lg",
   --pf-c-form__label--FontSize: var(--pf-global--FontSize--sm);
   --pf-c-form__label--LineHeight: var(--pf-global--LineHeight--sm);
   --pf-c-form__label--m-disabled--Color: var(--pf-global--disabled-color--100);
+  --pf-c-form__label--Cursor--hover: pointer;
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:70]: With component vars, the state (`hover`) in the var will be after the element the state applies to, and before the property that changes for that state. It follows the format of "when a `__label` is `hover`ed, set the `Cursor`"

More on that here - https://www.patternfly.org/v4/developer-resources/global-css-variables#component-variables

```suggestion
  --pf-c-form__label--hover--Cursor: pointer;
```

File: src/patternfly/components/Form/form.scss [hunk 2]
```diff
@@ -67,6 +67,8 @@ $pf-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "lg",
...
+  --pf-c-form__label--m-disabled--Cursor--hover: not-allowed;
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:71]:
```suggestion
  --pf-c-form__label--m-disabled--hover--Cursor: not-allowed;
```

File: src/patternfly/components/Form/form.scss [hunk 3]
```diff
@@ -268,15 +270,15 @@ $pf-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "lg",
   }
 
   &:not(.pf-m-disabled):hover {
-    cursor: pointer;
+    cursor: var(--pf-c-form__label--Cursor--hover);
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:273]:
```suggestion
    cursor: var(--pf-c-form__label--hover--Cursor);
```

File: src/patternfly/components/Form/form.scss [hunk 4]
```diff
@@ -268,15 +270,15 @@ $pf-c-form--m-horizontal--breakpoint-map: build-breakpoint-map("sm", "md", "lg",
...
   }
 
   &.pf-m-disabled {
     color: var(--pf-c-form__label--m-disabled--Color);
   }
 
   &.pf-m-disabled:hover {
-    cursor: not-allowed;
+    cursor: var(--pf-c-form__label--m-disabled--Cursor--hover);
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:281]:
```suggestion
    cursor: var(--pf-c-form__label--m-disabled--hover--Cursor);
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Hey @dgdavid! Thanks for the PR! It looks great - just one nit on the var names.

File: src/patternfly/components/Form/form.scss (hunk 1 above)
Comment (dgdavid) [reply src/patternfly/components/Form/form.scss:70]: @mcoker! Thanks a lot! My bad, I followed my intuition instead read the documentation you linked above. Excuse me :innocent:

**Review by mcoker — APPROVED** [review]
Nice!! Thanks @dgdavid!!

**Pull request merged**